static/src/javascripts/lib/robust.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+
+/*
+    Swallows (and reports) exceptions. Designed to wrap around modules at the "bootstrap" level.
+    For example "comments throwing an exception should not stop auto refresh"
+ */
+
+import reportError from 'lib/report-error';
+
+function catchErrors(fn: Function): ?Error {
+    let error;
+
+    try {
+        fn();
+    } catch (e) {
+        error = e;
+    }
+
+    return error;
+}
+
+function log(name: string, error: Error, customReporter?: Function): void {
+    let reporter = reportError;
+
+    if (customReporter) {
+        reporter = customReporter;
+    }
+
+    if (window.console && window.console.warn) {
+        window.console.warn('Caught error.', error.stack);
+    }
+
+    reporter(error, { module: name }, false);
+}
+
+function catchErrorsAndLog(name: string, fn: Function, reporter?: Function) {
+    const error = catchErrors(fn);
+
+    if (error) {
+        log(name, error, reporter);
+    }
+}
+
+function catchErrorsAndLogAll(modules: Array<any>) {
+    modules.forEach(pair => catchErrorsAndLog(pair[0], pair[1]));
```

Comment:
Yes exactly. To take an example, many promise libraries offered convenience methods to spread arrays and objects and spare you the trouble of manual destructuring. But now that we have it for free, we can just do it ourselves. I find it especially useful when we use arrays/objects as tuples rather than as collections.